foursight_core/react/api/checks.py

```diff
@@ -290,35 +402,87 @@ def _annotate_checks_with_schedules_from_lambdas(checks: dict, lambdas: dict) ->
-                    check_item["registered_file"] = check_decorator.get("file")
-                    check_item["registered_line"] = check_decorator.get("line")
-                    check_item["registered_module"] = check_decorator.get("module")
-                    check_item["registered_package"] = check_decorator.get("package")
-                    check_item["registered_github_url"] = check_decorator.get("github_url")
-                    check_decorator_kwargs = check_decorator.get("kwargs")
-                    if check_decorator_kwargs:
-                        check_item["registered_kwargs"] = check_decorator_kwargs
-                    check_decorator_args = check_decorator.get("args")
-                    if check_decorator_args:
-                        check_item["registered_args"] = check_decorator_args
+            for check_function_name in registered_checks:
+                if check_name == check_function_name:
+                    registered_check = checks_registry[check_function_name]
+                    check_item["registered_file"] = registered_check.get("file")
+                    check_item["registered_line"] = registered_check.get("line")
+                    check_item["registered_module"] = registered_check.get("module")
+                    check_item["registered_package"] = registered_check.get("package")
+                    check_item["registered_github_url"] = registered_check.get("github_url")
+                    registered_check_kwargs = registered_check.get("kwargs")
+                    if registered_check_kwargs:
+                        check_item["registered_kwargs"] = registered_check_kwargs
+                    registered_check_args = registered_check.get("args")
+                    if registered_check_args:
+                        check_item["registered_args"] = registered_check_args
+                    # Get any associated action; this is from the (new as of early December 2022)
+                    # action property specified in the @check_function decorator for the check
+                    # function; this is used only for informational purposes in the UI. 
+                    registered_action_function_name = registered_check.get("action")
+                    if registered_action_function_name:
+                        registered_action = checks_registry.get(registered_action_function_name)
+                        if registered_action and registered_action.get("kind") == "action":
+                            check_item["registered_action"] = {
+                                "name": registered_action.get("name"),
+                                "file": registered_action.get("file"),
+                                "line": registered_action.get("line"),
+                                "module": registered_action.get("module"),
+                                "package": registered_action.get("package"),
+                                "github_url": registered_action.get("github_url"),
```

Comment:
These github URLs assume a tagged commit - we should take care to ensure this tag is always propagated, otherwise this will link to nowhere.